iceoryx_hoofs/test/moduletests/test_cxx_variant.cpp

```diff
@@ -128,26 +129,48 @@ bool variant_Test::DTorTest::dtorWasCalled = false;
 int variant_Test::DoubleDelete::dtorCalls = 0;
 int variant_Test::DoubleDelete::ctorCalls = 0;
 
-TEST_F(variant_Test, DefaultCTor)
+TEST_F(variant_Test, DefaultCTorCreatesInvalidVariant)
 {
     EXPECT_THAT(sut.index(), Eq(iox::cxx::INVALID_VARIANT_INDEX));
 }
 
-TEST_F(variant_Test, emplaceValidElement)
+TEST_F(variant_Test, InitializedVariantReturnsCorrectIndex)
 {
-    ASSERT_THAT(sut.emplace<ComplexClass>(123, 456.789f), Eq(true));
+    sut.emplace<float>(1231.22F);
+    EXPECT_THAT(sut.index(), Eq(1U));
+}
+
+TEST_F(variant_Test, CreatingVariantFromPlainTypeReturnsProvidedValueAndCorrectIndex)
+{
+    iox::cxx::variant<ComplexClass, float> sut2{42.42F};
+
+    EXPECT_THAT(*sut2.get<float>(), Eq(42.42F));
+    EXPECT_THAT(sut2.index(), Eq(1U));
+}
+
+TEST_F(variant_Test, CreatingVariantFromLValueReturnsProvidedValueAndCorrectIndex)
+{
+    std::string string("Buhh");
+    iox::cxx::variant<std::string, float> sut2{string};
```

Comment:
A `cxx::variant<std::string, float> sut2{"Buhh"}` is not possible because the type `char[]` is directly passed to `emplace<char[]>` which checks the validity of the type without considering implicit conversions. IMHO we don't need this, happy to add it if we find it useful.

Reply:
I think for such cases the ctor could be called explicitly like `iox::cxx::variant<std::string, float> sut2{std::String{"Meh"}};` ... btw, would `iox::cxx::variant<std::string, float> sut2{{"Yay"}};`  work?

I guess this is not only related to strings but also to other types like having a `Foo(int, int)`?

Reply:
Nope, `sut2{{"Buhh"}}` does not work. The template type deduction in C++14 is still limited: `couldn't deduce template parameter ‘T’`. `sut2{std::String{"Buhh"}}` works. It's like that for all types.